dist/tools/mkconstfs/mkconstfs2.py

```diff
@@ -0,0 +1,176 @@
+#!/usr/bin/env python3
+
+import sys
+import argparse
+import pathlib
+import posixpath
+import io
+import itertools
+import mmap
+import shutil
+from binascii import hexlify
+
+C_HEADER = """/* This file was automatically generated by mkconstfs2.
+ * !!!! DO NOT EDIT !!!!!
+ */
+
+#include <stdint.h>
+#include "fs/constfs.h"
+
+"""
+
+FILE_TEMPLATE = """ {{
+  .path = "{target_name}",
+  .data = {mangled_name},
+  .size = sizeof({mangled_name})
+ }},
+"""
+
+C_FOOTER = """
+static const constfs_t _fs_data = {{
+    .files = _files,
+    .nfiles = sizeof(_files) / sizeof(_files[0]),
+}};
+
+vfs_mount_t {constfs_name} = {{
+    .fs = &constfs_file_system,
+    .mount_point = "{mount_pount}",
+    .private_data = (void *)&_fs_data,
+}};
+"""
+
+FILES_DECL = """
+static const constfs_file_t _files[] = {
+"""
+
+BLOB_DECL = """
+/** {fname} **/
+static const uint8_t {varname}[] = {{
+"""
+
+
+def _relpath_p(path, start):
+    return posixpath.relpath(path.as_posix(), start.as_posix())
+
+
+def mkconstfs(files, root_path, mount_point, constfs_name):
+    """Generate a C file containing a constant file system
+
+    Return
+    ------
+
+    chunks: Iterator yielding fragments of the of the output file.
+    """
+
+    filemap = {f: _relpath_p(f, root_path) for f in files}
+
+    yield C_HEADER
+    yield from itertools.chain.from_iterable(
+                print_file_data(*fl) for fl in filemap.items())
+
+    yield FILES_DECL
+
+    yield from (FILE_TEMPLATE.format(target_name=addroot(relp),
+                                     mangled_name=mangle_name(relp))
+                for relp in sorted(filemap.values()))
+
+    yield "};\n"
+
+    yield C_FOOTER.format(constfs_name=constfs_name, mount_pount=mount_point)
+
+
+def addroot(fname):
+    return "/" + fname if not fname.startswith("/") else fname
```

Comment:
Is it possible to rework this to something using the python `os.path` module or the `pathlib` module to keep it OS agnostic. It might just be what is causing issues for @MichelRottleuthner

Reply:
I think this won't help (for everything) because I also got backslashes in the middle of the path

Reply:
Ah, so do I understand correctly that the problem is that a path provided as `path\to\file` on windows shows up in the constfs as `/path\to\file`?

Reply:
Yes

Reply:
Ourch, that is not supposed to happen, but it's not because of `addroot`.

Reply:
Yeah, I misunderstood the initial issue. @jcarrano I think it might be solved by doing a recursive os.path.split() and `PosixPath.join()` on the supplied file

Reply:
Just had another look. The problem actually occurs only if relative paths are used. It then not only leaves backslashes there but also creates some crazy paths like `/../C:\path\to/C:/path/to/file`. I played around with various os.path / pathlib methods and found the following fix to make it working on linux and windows in the same way:
```diff
--- a/dist/tools/mkconstfs/mkconstfs2.py
+++ b/dist/tools/mkconstfs/mkconstfs2.py
@@ -1,5 +1,6 @@
 #!/usr/bin/env python3
 
+import os
 import sys
 import argparse
 import pathlib
@@ -50,7 +51,8 @@ static const uint8_t {varname}[] = {{
 
 
 def _relpath_p(path, start):
-    return posixpath.relpath(path.as_posix(), start.as_posix())
+    return posixpath.relpath(pathlib.Path(os.path.abspath(path)).as_posix(),
+                             pathlib.Path(os.path.abspath(start)).as_posix())
```

Reply:
@MichelRottleuthner Great! Can you push it to my branch, please?

Reply:
just tried that but somehow it doesn't work. Did you allow edits for maintainers?

Reply:
It was enabled. If it still does not work, I can apply a patch.
![screenshot_2018-06-29_11-40-18](https://user-images.githubusercontent.com/3057851/42085613-3cb792aa-7b91-11e8-833c-698f37a50e8b.png)

Reply:
nope still doesn't work just add it yourself, please